src/Components/Login/signup.js

```diff
@@ -0,0 +1,162 @@
+import React, { useState } from "react";
+import Avatar from "@material-ui/core/Avatar";
+import Button from "@material-ui/core/Button";
+import CssBaseline from "@material-ui/core/CssBaseline";
+import TextField from "@material-ui/core/TextField";
+import Link from "@material-ui/core/Link";
+import Grid from "@material-ui/core/Grid";
+import Typography from "@material-ui/core/Typography";
+import { makeStyles } from "@material-ui/core/styles";
+import Container from "@material-ui/core/Container";
+import empty from "is-empty";
+import { AccountCircle } from "@material-ui/icons";
+import { auth, generateUserDocument } from "../../firebaseCredentials.js";
+
+const useStyles = makeStyles((theme) => ({
+    paper: {
+        marginTop: theme.spacing(8),
+        display: "flex",
+        flexDirection: "column",
+        alignItems: "center",
+    },
+    avatar: {
+        margin: theme.spacing(1),
+        backgroundColor: theme.palette.secondary.main,
+    },
+    form: {
+        width: "100%",
+        marginTop: theme.spacing(1),
+    },
+    submit: {
+        margin: theme.spacing(3, 0, 2),
+    },
+}));
+
+export default function Signup() {
+    const classes = useStyles();
+
+    const [email, setEmail] = useState("");
+    const [password, setPassword] = useState("");
+    const [displayName, setDisplayName] = useState("");
+    const [allErrors, setAllErrors] = useState({});
+
+    function isNonEmptyForm() {
+        return email.length > 0 && password.length > 0;
+    }
+
+    async function createAccount(event) {
+        event.preventDefault();
+        const errors = {};
+
+        try {
+            const { user } = await auth.createUserWithEmailAndPassword(
+                email,
+                password
+            );
+            generateUserDocument(user, { displayName });
+            errors.login = "";
+        } catch (error) {
+            errors.login = "Email or password is incorrect.";
```

Comment:
Don't think this is needed from a signup perspective

Reply:
Good point, removed